internal/ast/choice.go

```diff
@@ -62,10 +74,41 @@ func (choice *Choice) Evaluate(p *Package) error {
 	}
 	for _, choiceCase := range choice.Cases {
 		for _, condition := range choiceCase.Conditions {
+			// in case the choice case is just a text string, replace that
+			//  by a dot expression with the choice selector type, so that the
+			// symbol is unique. If this is not done, the text value might be
+			// ambiguous, for example if the value  is defined in more than one enum.
+			// e.g. VALUE -> EnumType.VALUE
+			if condition.Condition.Type == parser.ZserioParserID {
+				// convert the selector from the parameter type to the underlying
+				// actual type (e.g. enum, bitfield)
+				parameterType, ok := choice.SelectorExpression.ResultSymbol.Symbol.(*Parameter)
+				if !ok {
+					return errors.New("choice selector type is not a parameter")
+				}
+				dotExpression := &Expression{
+					Type: parser.ZserioParserDOT,
+					Operand1: &Expression{
+						Type: parser.ZserioParserID,
+						Text: parameterType.Type.Name,
+					},
+					Operand2: condition.Condition,
+				}
+				condition.Condition = dotExpression
+			}
```

Comment:
Could this perhaps be a separate function?

Reply:
Done